js/ParticipantTopDonor.js

```diff
@@ -0,0 +1,43 @@
+$(document).ready(function () {
+    // UpdateInfo immediately, and then once every 10 seconds.
+    UpdateInfo();
+    window.setInterval(UpdateInfo, 10000);
+});
+
+function GetTopDonor(data) {    
+    var groupedData = {};
+    $.each(data, function(i, val) {
+        var name = val.donorName;
+        var amount = val.donationAmount;
+        if (groupedData.hasOwnProperty(name)) {
+            groupedData[name] += amount;
+        } else {
+            groupedData[name] = amount;
+        }
+    });
+
+    var sortedData = [];
+    for (var k in groupedData) sortedData.push([k, groupedData[k]]);
```

Comment:
if it's an array of objects, then I need to search the array for the right name each time I want to add. A donor can donate many different times, and will show up multiple times in the list of donations.